File: chs-frontend/src/services/courseService.js
import axios from 'axios';

const API_URL = '/api/courses';

const getAllCourses = () => {
  return axios.get(API_URL);
};

const getCourseById = (id) => {
  return axios.get(`${API_URL}/${id}`);
};

const getLessonsByCourseId = (courseId) => {
  return axios.get(`${API_URL}/${courseId}/lessons`);
};

const courseService = {
  getAllCourses,
  getCourseById,
  getLessonsByCourseId,
};

export default courseService;

File: src/main/java/com/chs/backend/services/CourseService.java
package com.chs.backend.services;

import com.chs.backend.models.Course;
import com.chs.backend.models.Lesson;
import com.chs.backend.repositories.CourseRepository;
import com.chs.backend.repositories.LessonRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
public class CourseService {

    @Autowired
    private CourseRepository courseRepository;

    @Autowired
    private LessonRepository lessonRepository;

    public List<Course> getAllCourses() {
        return courseRepository.findAll();
    }

    public Optional<Course> getCourseById(Long id) {
        return courseRepository.findById(id);
    }

    @Transactional(readOnly = true)
    public List<Lesson> getLessonsByCourseId(Long courseId) {
        return lessonRepository.findByCourseIdOrderByLessonOrderAsc(courseId);
    }

    // Additional methods for creating/updating courses and lessons can be added later
}
